descartes_planner/src/sparse_planner.cpp

```diff
@@ -23,11 +23,28 @@
  */
 
 #include <descartes_planner/sparse_planner.h>
-#include <boost/uuid/uuid_io.hpp>
 #include <algorithm>
 
 using namespace descartes_core;
 using namespace descartes_trajectory;
+
+namespace
+{
+    std::vector<descartes_core::TimingConstraint>
+    extractTiming(const std::vector<descartes_core::TrajectoryPtPtr>& points)
+    {
+      std::vector<descartes_core::TimingConstraint> result;
+      result.reserve(points.size());
+
+      for (const auto& ptr : points)
+      {
+        result.push_back(ptr->getTiming());
+      }
+      return result;
+    }
+}
+
+
```

Comment:
I hesitate to make general comments on compiler performance without benchmarking, but I doubt that output arguments are any faster because of return-value-optimizations and the fact that in c++-11 values returned from functions are implicitly moved.